docs/building-blocks/extending-templates/conditional-linking.md

```diff
@@ -0,0 +1,173 @@
+
+The problem is that resource manager does not include functionality to conditionally link templates using parameters. But there is a way to implement this using existing resource manager functions.
+
+In the following template, a variable includes an object with two or more properties. The two or more properties specify the `uri` of a template. Then, a `Microsoft.Resources/deployments` resource type is specified with a `templateLink` property whose `uri` is the variable object indexed by a parameter value. Another template specifies a value for the parameter, and during deployment this parameter value changes the index and this in turn changes the `uri`. 
+
+```json
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "condition": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "conditions": {
+      "no": "[uri(deployment().properties.templateLink.uri, '../Parameters/noDeploy.parameters.json')]",
+      "yes": "[uri(deployment().properties.templateLink.uri, '../Parameters/deploy.parameters.json')]"
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-01-01",
+      "type": "Microsoft.Resources/deployments",
+      "name": "[concat('conditional-resource-',parameters('condition'))]",
+      "properties": {
+        "mode": "Incremental",
+        "parameters": {},
+        "templateLink": { "uri": "[variables('conditions')[tolower(parameters('condition'))]]" }
+
+      }
+    }
+  ],
+  "outputs": {}
+}
+```
+
+During deployment, the `condition` parameter is typically specified in a parameter file as follows:
+
+```json
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "condition": { "value": "yes" }
+  }
+}
+```
+If `condition` is set to `yes`, the template named `deploy.parameters.json` is linked. If `condition` is set to `no`, the template named `noDeploy.parameters.json` is linked. Note that while there two conditions here, you can include as many values as you require and can link to any number of templates. You can use conditions `1`, `2`, and `3` to link to three different templates, or you can use conditions `AzureDNS`, `externalDNS`, or `internalDNS` to link to templates that specify the appropriate deployment.
+
+It's also possible to implement `optional` linking using some creative math. The modulo operation represents the remainder after dividing a dividend by a divisor. One of the properties of the modulo operation is that for any given `n`, `(n+2) mod (n+1)` equals `1` for every value of `n` greater than `0`. By substituting the length of an array or a string for `n` the result is a boolean operator for the presence of either. If the result is `0`, you can link to a template that doesn't deploy anything and if it's `1` you can link to a template that deploys the intended resource. 
```

Comment:
'for the presence of either' - is this testing for the presence of a string/array, or testing for length > 0 ? (i.e., does it handle nulls?)